pkg/cache/cache_test.go

```diff
@@ -32,16 +33,33 @@ func TestSaCache(t *testing.T) {
 	cache.addSA(testSA)
 
 	role, aud, useRegionalSTS, tokenExpiration = cache.Get("default", "default")
-	if role != roleArn {
-		t.Errorf("Expected role to be %s, got %s", roleArn, role)
-	}
-	if aud != "sts.amazonaws.com" {
-		t.Errorf("Expected aud to be sts.amzonaws.com, got %s", aud)
-	}
-	if useRegionalSTS {
-		t.Error("Expected regional STS to be true, got false")
+
+	assert.Equal(t, roleArn, role, "Expected role to be %s, got %s", roleArn, role)
+	assert.Equal(t, "sts.amazonaws.com", aud, "Expected aud to be sts.amzonaws.com, got %s", aud)
+	assert.True(t, useRegionalSTS, "Expected regional STS to be true, got false")
+	assert.Equal(t, int64(3600), tokenExpiration, "Expected token expiration to be 3600, got %d", tokenExpiration)
+}
+
+func TestNonRegionalSTS(t *testing.T) {
+	testSA := &v1.ServiceAccount{}
+	testSA.Name = "default"
+	testSA.Namespace = "default"
+	roleArn := "arn:aws:iam::111122223333:role/s3-reader"
+	testSA.Annotations = map[string]string{
+		"eks.amazonaws.com/role-arn":               roleArn,
+		"eks.amazonaws.com/sts-regional-endpoints": "false",
+		"eks.amazonaws.com/token-expiration":       "3600",
 	}
-	if tokenExpiration != 3600 {
-		t.Errorf("Expected token expiration to be 3600, got %d", tokenExpiration)
+
+	cache := &serviceAccountCache{
```

Comment:
IMO we need to test every case. 

1. annotation missing
2. annotation true
3. annotatoin false
4. annotation invalid
5. 1-4 when defaultRegionalSTS is set true
6. 1-4 when defaultRegionalSTS is set false

you can pass a fake client to New and put the testSa in it.